src/core/NavBar/NavBarLink.tsx

```diff
@@ -0,0 +1,60 @@
+import { NavLink } from 'react-router-dom';
+
+import NavBarLinkFlyout from './NavBarLinkFlyout';
+
+interface NavBarLinkProps {
+  to: string;
+  text: string;
+  icon: React.ReactNode;
+  expanded: boolean;
+  onActive: () => void;
+}
+
+export default function NavBarLink({
+  to,
+  icon,
+  text,
+  expanded,
+  onActive
+}: NavBarLinkProps) {
+  return (
+    <NavLink
+      to={to}
+      className={`relative h-12 shadow-sm group ${
+        expanded ? 'w-full' : 'w-12'
+      }`}>
+      {({ isActive }) => {
+        if (isActive) {
+          setTimeout(() => {
+            onActive();
+          }, 0);
+        }
```

Comment:
This looks a little bit strange, why do we have to do the timeout here?

Reply:
This is a bit tricky. We can't call `onActive` function without adding a timeout because it will cause the `NavBarLink` component re-render and I will get the following error in the console:
<img width="900" alt="image" src="https://github.com/AtmosFears/AtmosFears-FE/assets/52978053/6019204b-ac33-4fac-b74a-450b4142b270">

This `setTimeout` function makes a trick and the `onActive` function call is waiting in the callback queue. Because of that, `setTimeout` will be executed after all pending code was executed (e.g. rendering component in this example).

Maybe there is a better way to achieve the same effect but I haven't found a better way to get information whether the current `NavLink` is active in the React Router docs. Another solution might be to add navigation listener watching route changes but this will require writing more code and checking which route refers to which icon in the nav-bar. The current solution, even though it's quite strange, seems to be the simplest.